metrics/web/test/dashboard/domain/entities/metrics/receive_build_day_project_metrics_updates_test.dart

```diff
@@ -0,0 +1,308 @@
+
+        final metricsStream = useCase.call(projectIdParam);
+
+        final performanceMetricMatcher = predicate<BuildDayProjectMetrics>(
+          (metrics) => metrics.performanceMetric != null,
+        );
+
+        expect(
+          metricsStream,
+          emits(buildDayMetricsMatcher(
+            performanceMetricMatcher: performanceMetricMatcher,
+          )),
+        );
+      },
+    );
+
+    test(
+      ".call() returns a build day project metrics with the performance metric having zero average build duration if the returned build days have no builds",
+      () async {
+        final buildDay = createBuildDay();
+        final buildDays = [buildDay];
+        const expectedAverageDuration = Duration.zero;
+
+        whenStreamBuildDays().thenAnswer((_) => Stream.value(buildDays));
+
+        final metricsStream = useCase.call(projectIdParam);
+
+        final performanceMetricMatcher = predicate<BuildDayProjectMetrics>(
+          (metrics) {
+            final performanceMetric = metrics.performanceMetric;
+            final averageDuration = performanceMetric.averageBuildDuration;
+
+            return averageDuration == expectedAverageDuration;
+          },
+        );
+
+        expect(
+          metricsStream,
+          emits(buildDayMetricsMatcher(
+            performanceMetricMatcher: performanceMetricMatcher,
+          )),
+        );
+      },
+    );
+
+    test(
+      ".call() returns a build day project metrics with the performance metric having the average build duration equal to the total builds' duration divided by the total number of builds performed during the returned build days",
+      () async {
+        const firstDayDuration = Duration(seconds: 2);
+        const secondDayDuration = Duration(seconds: 3);
+        final firstBuildDay = createBuildDay(
+          totalDuration: firstDayDuration,
+          successful: 2,
+        );
+        final secondBuildDay = createBuildDay(
+          totalDuration: secondDayDuration,
+          failed: 3,
+        );
+        final buildDays = [firstBuildDay, secondBuildDay];
+        final totalDuration = firstDayDuration + secondDayDuration;
+        const totalNumberOfBuilds = 2 + 3;
+        final expectedAverageDuration = totalDuration ~/ totalNumberOfBuilds;
+
+        whenStreamBuildDays().thenAnswer((_) => Stream.value(buildDays));
+
+        final metricsStream = useCase.call(projectIdParam);
+
+        final performanceMetricMatcher = predicate<BuildDayProjectMetrics>(
+          (metrics) {
+            final performanceMetric = metrics.performanceMetric;
+            final averageDuration = performanceMetric.averageBuildDuration;
+
+            return averageDuration == expectedAverageDuration;
+          },
+        );
+
+        expect(
+          metricsStream,
+          emits(buildDayMetricsMatcher(
+            performanceMetricMatcher: performanceMetricMatcher,
+          )),
+        );
```

Comment:
Example.
If apply the suggestion from the above (with metrics matcher), we can write as follows:
```suggestion
        final metricsMatcher = buildDayMetricsMatcher(
          performanceMetricMatcher: isA<PerformanceMetric>().having(
            (metric) => metric.averageBuildDuration,
            'averageBuildDuration',
            equals(expectedAverageDuration),
          ),
        );

        expect(metricsStream, emits(metricsMatcher));
```